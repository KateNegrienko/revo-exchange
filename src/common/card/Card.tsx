import { FC } from "react";
import cn from "classnames";
import { ICardProps } from "./Card.interface";
import theme from "./Card.module.scss";

const Card: FC<ICardProps> = ({ children, className }) => (
  <div className={cn(theme.card, className)}>{children}</div>
);

export default Card;
