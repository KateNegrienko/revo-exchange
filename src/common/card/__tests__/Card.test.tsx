import React from "react";
import { render, shallow } from "enzyme";
import Card from "../Card";

describe("[Card snapshots]", () => {
  it("Should render properly with only required props", () => {
    const children = <div>children</div>;
    expect(render(<Card>{children}</Card>)).toMatchSnapshot();
  });

  it("Should add className to card component", () => {
    const testclass = "test";
    const wrap = shallow(<Card className={testclass} />);

    expect(wrap.hasClass(testclass)).toEqual(true);
  });
});
