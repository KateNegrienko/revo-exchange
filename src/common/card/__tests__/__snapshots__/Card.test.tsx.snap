// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Card snapshots] Should render properly with only required props 1`] = `
<div
  class="card"
>
  <div>
    children
  </div>
</div>
`;
